### PR TITLE
Add PU tag to Read Groups for all mapping methods

### PR DESCRIPTION
**Pull request by IdoBar**
Add PU tag to Read Groups for all mapping methods.
This is an improvement to PR #656

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
 - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/eager/tree/master/.github/CONTRIBUTING.md)
 - [ ] If necessary, also make a PR on the nf-core/eager _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint .`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by jfy133** [comment]
Looks good to me! I've just made a pull request into your new patch again, so you have the correct CHANGELOG file to update. Let me know if you need help updating the changelog.

**Comment by jfy133** [comment]
Ok, now you have the CHANGELOG for the next release,

please edit the [CHANGELOG file](https://github.com/IdoBar/eager/blob/patch-2/CHANGELOG.md), and under `2.3.1dev [unreleased]` and `fixed`, add something like

[#655 ](https://github.com/nf-core/eager/issues/655) - Updated read groups for all mappers to allow proper GATK validation

(or w/e).

Finally, please add your name and github handle to the [README](https://github.com/IdoBar/eager/blob/patch-2/README.md) under 'additional contributors' :).

Note all links above point to your patch branch of your fork, so you can edit those directly!

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ### `Fixed`
 
+ - [#655](https://github.com/nf-core/eager/issues/655) - Updated read groups for all mappers to allow proper GATK validation
```
Comment (IdoBar) [inline CHANGELOG.md:12]:
```suggestion
- [#655](https://github.com/nf-core/eager/issues/655) - Updated read groups for all mappers to allow proper GATK validation
```

**Review by apeltzer — APPROVED** [review]
Thank you @IdoBar

**Review by jfy133 — APPROVED** [review]

**Pull request merged**